share/spack/templates/container/Dockerfile

```diff
@@ -14,7 +14,7 @@ RUN mkdir {{ paths.environment }} \
 {{ manifest }} > {{ paths.environment }}/spack.yaml
 
 # Install the software, remove unnecessary deps
-RUN {% if monitor.enabled %}--mount=type=secret,id=su --mount=type=secret,id=st{% endif %} cd {{ paths.environment }} && spack env activate . {% if not monitor.disable_auth %}&& export SPACKMON_USER=$(cat /run/secrets/su) && export SPACKMON_TOKEN=$(cat /run/secrets/st) {% endif %}&& spack install {% if monitor.enabled %}--monitor {% if monitor.prefix %}--monitor-prefix {{ monitor.prefix }} {% endif %}{% if monitor.tags %}--monitor-tags {{ monitor.tags }} {% endif %}{% if monitor.keep_going %}--monitor-keep-going {% endif %}{% if monitor.host %}--monitor-host {{ monitor.host }} {% endif %}{% if monitor.disable_auth %}--monitor-disable-auth {% endif %}{% endif %}--fail-fast && spack gc -y
+RUN {% if monitor.enabled %}--mount=type=secret,id=su --mount=type=secret,id=st{% endif %} cd {{ paths.environment }} && spack env activate . {% if monitor.enabled %}{% if not monitor.disable_auth %}&& export SPACKMON_USER=$(cat /run/secrets/su) && export SPACKMON_TOKEN=$(cat /run/secrets/st) {% endif %}{% endif %}&& spack install {% if monitor.enabled %}--monitor {% if monitor.prefix %}--monitor-prefix {{ monitor.prefix }} {% endif %}{% if monitor.tags %}--monitor-tags {{ monitor.tags }} {% endif %}{% if monitor.keep_going %}--monitor-keep-going {% endif %}{% if monitor.host %}--monitor-host {{ monitor.host }} {% endif %}{% if monitor.disable_auth %}--monitor-disable-auth {% endif %}{% endif %}--fail-fast && spack gc -y
```

Comment:
can we split this across several lines with backslashes?

Reply:
Yeah, do you want to make a suggestion for the locations so I do it right the first time? I sort of prefer the one line, especially with the templating, but I'm open to splitting it somewhere.

Reply:
Also, when this actually renders you don't see most of that logic, it's a clean single line. So that was why it's still on one line.

Reply:
Nevermind I just did it, I realized I could just use the && as guides. Here is without monitor:

```dockerfile
# Install the software, remove unnecessary deps
RUN cd /opt/spack-environment && \
    spack env activate . && \
    spack install --fail-fast && \
    spack gc -y
```
and with
```dockerfile
# Install the software, remove unnecessary deps
RUN --mount=type=secret,id=su --mount=type=secret,id=st cd /opt/spack-environment && \
    spack env activate . && export SPACKMON_USER=$(cat /run/secrets/su) && export SPACKMON_TOKEN=$(cat /run/secrets/st) && \
    spack install --monitor --monitor-prefix ms1 --monitor-host http://127.0.0.1 --fail-fast && \
    spack gc -y
```

Reply:
That's way more readable -- awesome.

Reply:
Agreed! I didn't remember it being this bad - after you commented I looked and said "oh, geez, he's right."

Reply:
The other thought I had was that you could build the arg list in `containerize.py` and stick a less complex set of variables in the template context -- e.g.:

```
RUN {% monitor.run_args %} cd /opt/spack-environment \
  && spack env activate . \
  && spack install --fail-fast {% monitor.install_args %} \
  && spack gc -y
```

where the various args would be `""` if monitor is not enabled.  The exports could go in `run_args` I think, e.g.:
```
RUN --mount=type=secret,id=su --mount=type=secret,id=st export SPACKMON_USER=$(cat /run/secrets/su) \
  && export SPACKMON_TOKEN=$(cat /run/secrets/st) \
  && cd /opt/spack/environment && ...
```

But I don't think this is needed now -- it looks better as-is.